core/database_util.go

```diff
@@ -164,6 +95,20 @@ func GetHeadFastBlockHash(db ethdb.Database) common.Hash {
 	return common.BytesToHash(data)
 }
 
+// GetHeadBlockNum retrieves the block number of the current canonical head block.
+func GetHeadBlockNum(db ethdb.Database) *big.Int {
```

Comment:
@karalabe What issue are you referring too? I'm not aware of any bugs associated with this method. Also, have you benchmarked whether this incurs a significant performance hit in practice? Note that leveldb itself caches queries, so multiple calls to get the "LastBock" will be cached.

The reason for this method is that for homestead protocol changes we need to check based on block number, and I couldn't find any methods for getting the block number of the current HEAD. If there is a better way to get the block number of the current HEAD block I'm all for that.